### ISSUE TITLE
`UnownedTaskExecutor` should be `Hashable`

### ISSUE THREAD

**Issue by fabianfett** [apple/swift#73780]
### Motivation

Swift 6 introduces TaskExecutors thanks to [SE-0417](https://github.com/apple/swift-evolution/blob/main/proposals/0417-task-executor-preference.md).

As part of SE-0417, the `withUnsafeCurrentTask` API was extended so that users can query the `unownedTaskExecutor`. This is very useful in performance sensitive applications. As I'm currently building a ConnectionPool, I would love to use the returned `UnownedTaskExecutor` to lookup connections that are running on the executor that asks for the connection to reduce context switches. 

For this it would be optimal if `UnownedTaskExecutor` not just implemented `Equatable` but also `Hashable`. So that I don't need to do a linear search through all potential executors and instead can use a dictionary.

### Proposed solution

Add `Hashable` conformance to `UnownedTaskExecutor`.

**Comment by ktoso** [comment]
Agreed, this should be uncontroversial. When used carefully like you do I can see the utility of that.

**Comment by ktoso** [comment]
rdar://128452304

**Comment by ktoso** [comment]
[rdar://128452304](rdar://128452304)
